app/ante/max_tx_size.go

```diff
@@ -0,0 +1,29 @@
+package ante
+
+import (
+	"github.com/celestiaorg/celestia-app/v3/pkg/appconsts"
+	v3 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v3"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MaxTxSizeDecorator ensures that a tx can not be larger than
+// application's configured versioned constant.
+type MaxTxSizeDecorator struct{}
+
+func NewMaxTxSizeDecorator() MaxTxSizeDecorator {
+	return MaxTxSizeDecorator{}
+}
+
+// AnteHandle implements the AnteHandler interface. It ensures that tx size is under application's configured threshold.
+func (d MaxTxSizeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// Tx size rule applies to app versions v3 and onwards.
+	if ctx.BlockHeader().Version.App < v3.Version {
+		return next(ctx, tx, simulate)
+	}
+
+	if len(ctx.TxBytes()) > appconsts.MaxTxBytes(ctx.BlockHeader().Version.App) {
```

Comment:
can we also delete the MaxTotalBlobSizeDecorator?

are there any scenarios where this context is not set properly? like in prepare proposal?

Reply:
I think we should still keep it because one limits by number of shares and the other number of bytes. I think you pointed out that you could have a PFB that is full of 1 byte blobs which would exceed the share limit but no the byte limit

Reply:
`MaxTotalBlobSizeDecorator` only applies to app version 1

```
		// Ensure that the tx's total blob size is <= the max blob size.
		// Only applies to app version == 1.
		blobante.NewMaxTotalBlobSizeDecorator(blobKeeper),
```

and since the ante handler added in this PR only applies to app version >= 3, I think we need to retain `MaxTotalBlobSizeDecorator`.

The ante handler that limits based on number of shares is `NewBlobShareDecorator` and that one is applied to app version >= 2.

```
		// Ensure that the blob shares occupied by the tx <= the max shares
		// available to blob data in a data square. Only applies to app version
		// >= 2.
		blobante.NewBlobShareDecorator(blobKeeper),
```